util/logger/src/lib.rs

```diff
@@ -106,6 +106,17 @@ macro_rules! log_enabled_target {
     };
 }
 
+#[macro_export(local_inner_macros)]
+macro_rules! metric {
+    ($class:expr, $context:expr) => {
```

Comment:
Better avoid the name `class`, which have too many meanings. 

For the metrics, I recommend that it should contains 3 kinds of data:

* label: which type of metric
* tags: used to filter metrics, for example, tx hash, peer id are good candidate of tags
° values: metric values, such as served requests, validated transaction counts

FYI:

- https://docs.influxdata.com/influxdb/v1.7/write_protocols/line_protocol_tutorial/
- https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md

A sample JSON may look like:

```json
{"metric":"p2p_requests", "tags":{"peer_id": "0"}, "values": {"requests": 1}, "timestamp": "..."}
```